sdc/rewrites/read_csv_consts.py

```diff
@@ -0,0 +1,161 @@
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+from numba.rewrites import register_rewrite, Rewrite
+from numba.ir_utils import find_callname, guard, mk_unique_var
+from numba import ir, errors, consts
+
+from sdc.rewrites.ir_utils import remove_unused_recursively
+
+
+def find_build_sequence(func_ir, var):
+    """Reimplemented from numba.ir_utils.find_build_sequence
+    Added 'build_map' to build_ops list.
+    """
+    from numba.ir_utils import (require, get_definition)
+
+    require(isinstance(var, ir.Var))
+    var_def = get_definition(func_ir, var)
+    require(isinstance(var_def, ir.Expr))
+    build_ops = ['build_tuple', 'build_list', 'build_set', 'build_map']
+    require(var_def.op in build_ops)
+    return var_def.items, var_def.op
+
+
+def _new_definition(func_ir, var, value, loc):
+    func_ir._definitions[var.name] = [value]
+    return ir.Assign(value=value, target=var, loc=loc)
+
+
+class ConstantInference(consts.ConstantInference):
+
+    def _infer_expr(self, expr):
+        if expr.op == 'build_map':
+            def inf_const(value):
+                return self.infer_constant(value.name, loc=expr.loc)
+            return {inf_const(k): inf_const(v) for k, v in expr.items}
+        return super()._infer_expr(expr)
+
+
+@register_rewrite('before-inference')
+class RewriteReadCsv(Rewrite):
+    """
+    Searches for calls of pandas.read_csv() and replace it with calls of read_csv.
+    """
+
+    _pandas_read_csv_calls = [
+        ('read_csv', 'pandas'),             # for calls like pandas.read_csv()
+        ('read_csv', 'pandas.io.parsers'),  # for calls like read_csv = pandas.read_csv, read_csv()
+    ]
+
+    _read_csv_const_args = ('names', 'dtype', 'usecols')
+
+    def match(self, func_ir, block, typemap, calltypes):
+        # TODO: 1. save instructions of build_map, build_list for read_csv params
+        # 2. check that vars are used only in read_csv
+        # 3. replace vars with build_tuple inplace
+
+        self.func_ir = func_ir
+        self.block = block
+        self.consts = consts = {}
+
+        # Find all assignments with a right-hand read_csv() call
+        for inst in block.find_insts(ir.Assign):
```

Comment:
you could use [find_operations(block, 'call')](https://github.com/IntelPython/sdc/blob/master/sdc/rewrites/ir_utils.py#L48) instead.

Reply:
FIxed. Use find_operations.